### PR TITLE
start/stop VM based on conditions script

### PR DESCRIPTION
**Pull request by belwase**

### CODE REVIEW

**Review by mahnouman — APPROVED** [review]

File: scripts/vm_utils.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    VSPHERE_HOST=
+    VSPHERE_USERNAME=
+    VSPHERE_PASSWORD=
+    mode_tag=start|shutdown
+    start_shut_tag=
+    not_shutdown_tag=
+    """
+
+    def __init__(self):
+        self.mode_tag = environ.get("mode_tag", None)
```
Comment (dinis-cruz-gw) [inline scripts/vm_utils.py:22]: please see comments in https://github.com/k8-proxy/k8-vmware/pull/24

File: scripts/vm_utils.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
+from os import environ
+from dotenv import load_dotenv
+from k8_vmware.vsphere.Sdk import Sdk
+from k8_vmware.vsphere.VM import VM
+
+load_dotenv()
```
Comment (dinis-cruz-gw) [inline scripts/vm_utils.py:6]: should not be here

File: scripts/vm_utils.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        if notes:
+            if self.not_shutdown_tag:
+                if self.not_shutdown_tag.lower() in notes:
+                    return False
+
+            if self.start_shut_tag.lower() in notes:
+                if self.mode_tag in ["start", "shutdown"]:
+                    altered = vm.info()["Name"]
+                    if self.mode_tag == "start":
+                        print("starting..", altered)
```
Comment (dinis-cruz-gw) [inline scripts/vm_utils.py:45]: refactor into methods that :
- calculate targets
- do actions
- log outcome

File: scripts/vm_utils.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+        if altered_vms:
+            print("Altered VMs: ")
+            print("=============")
+            print("\n".join(altered_vms))
+        else:
+            print("No VM was Altered !")
+
+
+VMUtils().process()
```
Comment (dinis-cruz-gw) [inline scripts/vm_utils.py:86]: main should only have one command
use tests to trigger execution

**Pull request merged**